demo/admin/src/shop/shopProductPage/SaveShopProductHandler.tsx

```diff
@@ -0,0 +1,42 @@
+import React, { createContext, useContext } from "react";
+
+type SaveShopProductHandlerContextType = {
+    registerHandleSubmit: any;
+    saveAll: () => Promise<void>;
+};
+
+export const SaveShopProductHandlerContext = createContext<SaveShopProductHandlerContextType | undefined>(undefined);
```

Comment:
sorry for commenting this WIP.

I thought about re-using the already existing Prompt that has a saveAction by using RouterContext (which is created by RouterPromptHandler.

That currenctly only has register/unregister but can relatively easily expanded to pass the registeredMessages thru the context. (saveActions could be merged into registeredMessages)

Reply:
No problem, thanks for the input. Your idea of reusing the existing Prompt with a saveAction using RouterContext created by RouterPromptHandler sounds promising. I'll try that, thanks!